src/Api/Controllers/NotificationController.cs

```diff
@@ -34,5 +36,16 @@ public IActionResult GetNotifications()
             else
                 return NotFound();
         }
+
+        // POST api/Notifications/:id/markread
+        [HttpPost("{id}")]        
+        public IActionResult Post(int userId)
```

Comment:
The commet describes :  // POST api/Notifications/:id/markread
But [HttpPost("{id}")] resolves to "POST api/Notifications/:id/" Which assumes that you put a new notification to the system

And it's not the userId you have to pass.